### PR TITLE
fix: throw instead of silently overwriting timers

### PR DESCRIPTION
**Pull request by lenaschoenburg**
## Description

Here we throw an exception when a timerId was generated that alreadyexists in the timerJobMap.
Throwing here will fail when scheduling a new timer or when resubmitting a recurring timer.

This should never happen, but we have one heapdump where it looks like it did: https://github.com/camunda-cloud/zeebe/issues/8776

## Related issues

relates to #8776

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

### CODE REVIEW

**Comment by oleschoenburg** [comment]
I think we should catch this exception in case re-submitting a recurring timer fails so that we don't kill an actor thread when this happens

File: util/src/main/java/io/camunda/zeebe/util/sched/ActorTimerQueue.java [hunk 1]
```diff
@@ -52,6 +49,13 @@ public void schedule(final TimerSubscription timer, final ActorClock now) {
 
     final long timerId = scheduleTimer(deadline);
     timer.setTimerId(timerId);
+    if (timerJobMap.containsKey(timerId)) {
+      throw new IllegalStateException(
+          "Failed scheduling, timer with id "
+              + timerId
+              + " already exists: "
+              + timerJobMap.get(timerId));
```
Comment (romansmirnov) [inline util/src/main/java/io/camunda/zeebe/util/sched/ActorTimerQueue.java:57]: I would suggest implementing `TimerSubscription#toString()` to gather additional information on the already scheduled timer. WDYT?

**Review by romansmirnov — CHANGES_REQUESTED** [review]
> I think we should catch this exception in case re-submitting a recurring timer fails so that we don't kill an actor thread when this happens

But I think we already have implemented a failure handling? Basically, the `IllegalStateException` is thrown which will be caught in `ActorJob#execute()` which may cause to fail the actor but to my understanding, the actor thread will keep running?

So, to me, the question would be if it is okay that the respective actor fails and what would be the impact of it?

What do you think?

File: util/src/main/java/io/camunda/zeebe/util/sched/ActorTimerQueue.java (hunk 1 above)
Comment (oleschoenburg) [reply util/src/main/java/io/camunda/zeebe/util/sched/ActorTimerQueue.java:57]: Makes sense, I added a toString that includes most properties. :+1:

**Review by romansmirnov — APPROVED** [review]
LGTM 🚀

**Comment by oleschoenburg** [comment]
bors retry

**Comment by oleschoenburg** [comment]
bors retry

**Comment by oleschoenburg** [comment]
bors retry

**Pull request merged**